Release/alphav0

ZzRanger opened the pull request:
## Overview

This PR will contain all the commits for the Planner launch at ACM kickoff tomorrow.

wbne committed:
feat(component): Added Course Audit

Provides a table allowing users to manually add courses they've taken. Not
implemented yet in any pages

re #46

wbne committed:
fix: Fixed importing data.js

Vercel did not like importing data.ts so it was changed to data.js

re #51

wbne committed:
fix: Changed data.js to data

Vercel does not like .js or .ts file extensions

re #51

wbne committed:
fix: Validate Types

Validated types in CourseAudit.tsx

Fix #50

wbne committed:
refactor: Add Requested Changes

Added the requested changes in issue #51

Re #51

wbne committed:
feat: Implement CourseAudit

Implemented Course Audit in the profile page in the bottom of the page

Re #51

wbne committed:
fix: Fix visual bugs and typing

Fixed sorting by name, styling errors, and type validation

Re #51

wbne committed:
fix: Fix changing size when selecting rows

When selecting a row, the table width would change due to a padding styling.

Fix #51

wbne committed:
fix: Allow same course name in multiple rows

Rows now have a unique identifier beyond naming scheme.
An error message now appears when trying to add an invalid or duplicate
course.
Default row header is now the course adding component.
Fixed formatting, comments, and variable names.

Fix #51

wbne committed:
fix: Fix redux-persist loading loop

Implemented PersistReducer to store.ts

wbne committed:
feat: Implement redux storage into CourseAudit

wbne committed:
refactor: Validate typing for changed files

Fix #51

wbne committed:
chore: Update packages

rajmeet2001 committed:
feat: Implement New Onboarding Flow

rajmeet2001 committed:
fix(logfiledeleted): A log file was removed

ZzRanger committed:
feat(onboarding): Add intro pages

ZzRanger committed:
refactor(onboarding): Add buttons to main onboarding

ZzRanger committed:
refactor(onboarding): Refactor disclaimer & privacy pages

The disclaimer & privacy pages in Onboarding_Pages are replaced with the
Disclaimer component

ZzRanger committed:
feat(onboarding): Add navigation display

ZzRanger committed:
refactor(onboarding): Clean up onboarding code

ZzRanger committed:
chore(onboarding): Change onboarding to match new Figma mockups

Revamped onboarding to match the mockups made by Lisa & Dylan. Onboarding now
has a navigational component at the top to inform the user of their progress.
Users can also now add their major & list more detailed information about the
scholarships & honors programs that they're in. Finally, an onboarding route was
added to _app.tsx

ZzRanger committed:
chore(onboarding): Small onboarding UI changes

ZzRanger committed:
chore(onboarding): Add form validation

ZzRanger committed:
chore(onboarding): Add page validation to Navigation component

ZzRanger committed:
feat(onboarding): Create transfer credit tool

ZzRanger committed:
docs(onboarding): Modify  documentation to onboarding pages

ZzRanger committed:
refactor(onboarding): Move hardcoded data values to dummy json file

ZzRanger committed:
chore(onboarding): Connect onboarding pages to app/onboarding.tsx

ZzRanger committed:
fix(onboarding): Fix DegreePicker bug

DegreePicker would occasionally not render correctly on pg_1.tsx (wrong values
were shown, extra components were added / removed)

ZzRanger committed:
chore(onboarding): Add animation to onboarding

The animation wasn't added correctly when all the onboarding pages were
connected to app/onboarding.tsx. This commit fixes that

ZzRanger committed:
refactor(onboarding): Create DegreePickerGallery component

Moved the implementation of DegreePicker from Onboarding_Pages/pg_1.tsx to
components/onboarding/DegreePickerGallery.tsx to reduce the complexity of
Onboarding_Pages/pg_1.tsx.

ZzRanger committed:
fix(onboarding): Fix bugs with Navigation and Onboarding_Pages/pg_2.tsx

ZzRanger committed:
refactor(onboarding): Create TransferCreditGallery component

Moved implementation of TransferCredit from Onboarding_Pages/pg_3.tsx to
components/onboarding/TransferCreditGallery.tsx to reduce complexity of
Onboarding_Pages/pg_3.tsx

ZzRanger committed:
fix(onboarding): Hide AppNavigation bar while in onboarding

ZzRanger committed:
fix(course audit): Fix build error

In components/planner/history/StudentHistory/CourseAudit.tsx, the
TablePagination component was missing property "onPageChange". This subsequently
threw an error whenever a production build of the app is attempted

ZzRanger committed:
feat(planner): Create CourseSelector component

This component is a sidebar that displays courses the user can take and  allows
the user to select courses to add into their degree planner.

ZzRanger committed:
feat(planner): Create ability to add courses to planner

After courses are selected and the button "Add Courses" is pushed inside
components/planner/CourseSelector.tsx, a Droppable is created inside
components/planner/DraggableItemContainer.tsx for the user to drag courses into
the planner. This commit has created the core logic to add courses. Styling of
the components & adding further non-essential logic sitll needs to be done and
will be addressed in future commits.

ZzRanger committed:
feat(planner): Add the ability for users to select courses

The component AddNewCourseDialog.tsx is a dialog screen that allows students to
search up their desired UTD course and add it to their degree planner. This
component currently lives inside components/planner/CourseSelector.tsx, but it
may be moved to a different location in the future. Currently, the dialog screen
needs proper styling, lacks certain interactive features, and is using dummy
data from data/dummy_planner_course_data.json. These issues need to be addressed
in future commits

ZzRanger committed:
feat(planner): Implement add/remove semesters in Degree Planner

ZzRanger committed:
feat(planner): Create dialog for generating a new degree plan

When a user is in "app/plans", there is now a button that opens a dialog for the
user to create a new degree plan. Once the user is finished filling out the
fields, the dialog will close and the user will be redirected to
"app/plans/[planId]". Most of the functionality in NewPlanDialog.tsx still needs
to be implemented, as well as generating the degree plan based on user input.

ZzRanger committed:
feat(planner): Create working Searchbar

Users now have the ability to filter through majors and courses in
NewPlanDialog.tsx and AddCourseDialog.tsx respectively. The filtering logic will
need to be adjusted once dummy data is replaced by real data. Additionally,
logic to support chips needs to be implemented in the future.

BREAKING CHANGE: type NewPlanFlowState: 'FINALIZE_PLAN_GENERATION' value was
removed.

ZzRanger committed:
chore: Update package-lock.json

ZzRanger committed:
refactor(planner): Clean up degree planner code

Moved the logic for adding courses into the useDraggItemContainer hook, data
into json files in the /data folder. Additionally, the data required to render
[...planId].tsx is loaded from a centralized function in [..planId].tsx called
loadData().

ZzRanger committed:
refactor(planner): Move data interfacing logic to custom react hooks

In "app/plans/[...planId].tsx, the process of obtaining the user's plan and
loading in the plan state has been moved to the usePlan hook. Additionally, the
logic for calling courseAttempts has been moved to its own file.

ZzRanger committed:
docs(planner): Add documentation to planner components

ZzRanger committed:
refactor(planner): Clean up planner files & make small UI changes

ZzRanger committed:
Merge branch 'feature/new-onboarding' into feature/degree-planner

ZzRanger committed:
refactor(planner): Add requested changes

ZzRanger committed:
feat(planner): Connect planner to Redux

Previously, the planner saved its plan data by writing directly to localstorage.
Now, all plan data is saved in Redux and persisted using Redux Persist.

ZzRanger committed:
chore(auth): Connect onboarding to application

ZzRanger committed:
chore(planner): Refactor Planner behavior for alpha release

ZzRanger committed:
feat(planner): Add ability to remove courses

ZzRanger committed:
feat(planner): Add ability to remove courses

wbne committed:
fix: Change onChangeRows to onRowChange due to deprecation

wbne committed:
feat: Link firebase storage with local redux stores

ZzRanger committed:
chore: Fix build errors

ZzRanger committed:
Merge remote-tracking branch 'remotes/origin/release/alphav0' into release/alphav0

wbne committed:
fix: Remove unnecessary import

ZzRanger committed:
fix(planner): Fix planner bugs

WillieCubed commented:
#### 📝 Changed routes:


| [**…rding_Pages/disclaimer**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/disclaimer) | [**…ding_Pages/disclaimers**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/disclaimers) |
|:-:|:-:|
|<a href="https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/disclaimer"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/c20s6nmFWWhqWTuzvhYR_QRKVm3uEpFOtl4EVGEKs3HpZ25fsgun8kXPV4q4jDT3bd4TaWA57c8NzE-grs5l-Q.png&w=600" alt="Screenshot of /Onboarding_Pages/disclaimer" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/x2hPmnt30sP3a70w759v5wpPDv3N3RVjJ-aqeNXIm8fdEEpH8u2c2kTR7xZaN3emzmDBOZAZsbH7DobWzO6kYw.png">(view full size)</a> |<a href="https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/disclaimers"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/gxwaZefQ4Jdi64WeCD41gI7O8jACdGivBG6OpZUAJSvtBFQR2jhM6Jl00LYROxESRm4tms0GBSFnw7EkUjkP2w.png&w=600" alt="Screenshot of /Onboarding_Pages/disclaimers" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/2Z3ykkw-NSRvuAz-ZImg-yv-i_V59bEghLug43iECHCOIMyvhjLXql65k-_I0fjCiTpglyKeKohLS248BSCSPg.png">(view full size)</a> |



| [**/Onboarding_Pages/pg_1**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/pg_1) | [**/Onboarding_Pages/pg_2**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/pg_2) |
|:-:|:-:|
|<a href="https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/pg_1"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/DW29JmtA71nFXkrC1iiTmodIoNnFjWWzF-zbP4lOy-Fs-jVSOli_E_dQlomImH1kHuy-WP6uBIDLYG2Fg7W0rQ.png&w=600" alt="Screenshot of /Onboarding_Pages/pg_1" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/xauJFqdTPjLLJrAUZp75tiuVEzt2c_0lX8o1hBQ-bYNU54VkhfnsLIUnAvmc60uM9RFFPRHRUoiKAsjcR1l0Cw.png">(view full size)</a> |<a href="https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/pg_2"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/bZH4e401LKJiaRCx0RZ1nIZB9L3Cy-Tq4ZvW6ewsS-v1f7Whhanp9295IaYVWhirWp353igSg8tUieHXmlJYtg.png&w=600" alt="Screenshot of /Onboarding_Pages/pg_2" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/nYWIb6O2FuaubpoBkebA_U9qnS6KRpdXGrIDUceWrUCCvAjwip9yt1ZKS8NhySl092BnLAWBDJo3TelmCoxNwQ.png">(view full size)</a> |



| [**/Onboarding_Pages/pg_3**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/pg_3) | [**/Onboarding_Pages/privacy**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/privacy) |
|:-:|:-:|
|<a href="https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/pg_3"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/k8pfLDHocNGpKKahs-q7S6vccMlaVoN8brJ5u9vdJEbH61swek0V0UrzV7uzzzu-xbwWdDuVmSOvmw2d298-eQ.png&w=600" alt="Screenshot of /Onboarding_Pages/pg_3" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/jk84iV1HA2_WvVNMZ7niNafG6_Tc4_noIIA5ZS1Mp_rnyK9Wn0u9b_naAS-6dtXhoiEZpCL09wFRO48mUiqqKw.png">(view full size)</a> |<a href="https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/privacy"><img src="https://images.weserv.nl?url=https://sjc.microlink.io/RgASk67fF9OqTjE2NQm9ymrdnsSs10IyHV0Q3XZaQGmy5yfpuq9haAKyoIo85H-E3gG_0XXEfbyTPy_jsW_0Yg.png&w=600" alt="Screenshot of /Onboarding_Pages/privacy" width="300"></a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/wT1ArnbvUvdpnK2Ahx4faOPKGwkMQqF7R_WfXmav4SNW38w0aAGbR16cTidaq-xCHGTP8aV4xif-xujitZjFhw.png">(view full size)</a> |



And 7 other routes:
- [**/Onboarding_Pages/welcome**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/Onboarding_Pages/welcome)
- [**/app/auth/\[task\]**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/app/auth/[task])
- [**/app/onboarding**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/app/onboarding)
- [**/app/plans/\[...planId\]**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/app/plans/[...planId])
- [**/app/plans**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/app/plans)
- [**/app/profile**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/app/profile)
- [**/**](https://nebula-planner-dlkjt9ia0-acmutd.vercel.app/)



Commit 1d94800083f957e3734cf0a54ba2b731fb3d0957 (https://nebula-planner-dlkjt9ia0-acmutd.vercel.app).

ZzRanger committed:
feat(planner): Connect planner to Nebula API

ZzRanger committed:
fix(planner): Fix CourseSelector bug

ZzRanger committed:
fix(planner): Planner bug fix

ZzRanger committed:
feat(home): Build temporary home page for kickoff

ZzRanger committed:
feat(planner): Add plan settings dialog

rajmeet2001 committed:
chore(two files in components/auth): Added styles for sign up and login

New Styling for sign up and login. Updated for launch

ZzRanger committed:
fix(planner): Fix planner saving

ZzRanger committed:
Merge remote-tracking branch 'remotes/origin/release/alphav0' into release/alphav0

ZzRanger committed:
chore(auth): Connect new auth page

ZzRanger committed:
Merge remote-tracking branch 'remotes/origin/feature/degree-planner' into release/alphav0

ZzRanger committed:
fix(planner): Fix Firebase redux integration

rajmeet2001 committed:
fix(signin file): Fixed the Sign In with Google Button

Went ahead and replaced the missing google image with text of Google

ZzRanger committed:
chore: Merge

ZzRanger committed:
fix: Fix

rajmeet2001 committed:
feat(signin and signup files): Added a new gradient background to signin

New gradient based background was added to both sign in and sign up files.
Addtionally, some fixes were added

ZzRanger committed:
fix(planner): Fix styling

ZzRanger committed:
Merge remote-tracking branch 'remotes/origin/release/alphav0' into release/alphav0

rajmeet2001 committed:
feat(logo): Planner Logo Added

Replaced Nebula Logo with Planner Logo

ZzRanger committed:
fix: UI planner fix

ZzRanger committed:
Merge remote-tracking branch 'remotes/origin/release/alphav0' into release/alphav0

ZzRanger committed:
feat(auth): Add redirect if user signed in

ZzRanger committed:
fix(auth): Remove alert in signin.tsx
